internal/encoding/merge/parquet.go

```diff
@@ -0,0 +1,295 @@
+	case "uint32":
+		col.SchemaElement.Type = parquet.TypePtr(parquet.Type_INT32)
+		col.SchemaElement.LogicalType = parquet.NewLogicalType()
+		col.SchemaElement.LogicalType.INTEGER = &parquet.IntType{BitWidth: 32, IsSigned: false}
+		col.SchemaElement.ConvertedType = parquet.ConvertedTypePtr(parquet.ConvertedType_UINT_32)
+		fieldHandler = uintHandler(32)
+	case "int64":
+		col.SchemaElement.Type = parquet.TypePtr(parquet.Type_INT64)
+		col.SchemaElement.LogicalType = parquet.NewLogicalType()
+		col.SchemaElement.LogicalType.INTEGER = &parquet.IntType{BitWidth: 64, IsSigned: true}
+		col.SchemaElement.ConvertedType = parquet.ConvertedTypePtr(parquet.ConvertedType_INT_64)
+		fieldHandler = intHandler(64)
+	case "uint64":
+		col.SchemaElement.Type = parquet.TypePtr(parquet.Type_INT64)
+		col.SchemaElement.LogicalType = parquet.NewLogicalType()
+		col.SchemaElement.LogicalType.INTEGER = &parquet.IntType{BitWidth: 64, IsSigned: false}
+		col.SchemaElement.ConvertedType = parquet.ConvertedTypePtr(parquet.ConvertedType_UINT_64)
+		fieldHandler = uintHandler(64)
+	case "float64":
+		col.SchemaElement.Type = parquet.TypePtr(parquet.Type_FLOAT)
+		fieldHandler = floatHandler
+	case "double":
+		col.SchemaElement.Type = parquet.TypePtr(parquet.Type_DOUBLE)
+		fieldHandler = doubleHandler
+	case "int":
+		col.SchemaElement.Type = parquet.TypePtr(parquet.Type_INT64)
+		col.SchemaElement.LogicalType = parquet.NewLogicalType()
+		col.SchemaElement.LogicalType.INTEGER = &parquet.IntType{BitWidth: 64, IsSigned: true}
+		col.SchemaElement.ConvertedType = parquet.ConvertedTypePtr(parquet.ConvertedType_INT_64)
+		fieldHandler = intHandler(64)
+	case "json":
+		col.SchemaElement.Type = parquet.TypePtr(parquet.Type_BYTE_ARRAY)
+		col.SchemaElement.LogicalType = parquet.NewLogicalType()
+		col.SchemaElement.LogicalType.JSON = &parquet.JsonType{}
+		col.SchemaElement.ConvertedType = parquet.ConvertedTypePtr(parquet.ConvertedType_JSON)
+		fieldHandler = jsonHandler
+	default:
+		return nil, nil, fmt.Errorf("unsupported type %q", typ)
+	}
+
+	fieldHandler = optionalHandler(fieldHandler)
+
+	return col, fieldHandler, nil
+}
+
+func byteArrayHandler(s interface{}) (interface{}, error) {
+	localString := fmt.Sprintf("%v", s)
+
+	return []byte(localString), nil
+}
+
+func booleanHandler(s interface{}) (interface{}, error) {
+	localString := fmt.Sprintf("%v", s)
+
+	return strconv.ParseBool(localString)
+}
+
+func uintHandler(bitSize int) func(interface{}) (interface{}, error) {
+	return func(s interface{}) (interface{}, error) {
+		localString := fmt.Sprintf("%v", s)
+		i, err := strconv.ParseUint(localString, 10, bitSize)
+		if err != nil {
+			return nil, err
+		}
+		switch bitSize {
+		case 8, 16, 32:
+			return uint32(i), nil
+		case 64:
+			return i, nil
+		default:
+			return nil, fmt.Errorf("invalid bit size %d", bitSize)
+		}
+	}
+}
+
+func intHandler(bitSize int) func(interface{}) (interface{}, error) {
+	return func(s interface{}) (interface{}, error) {
```

Comment:
Same comment, don't convert to string and parse, check the type and cast to the appropriate type, no need to do anything special here.